src/bson.c

```diff
@@ -1296,7 +1294,7 @@ void phongo_zval_to_bson(zval *data, php_phongo_bson_flags_t flags, bson_t *bson
 
 				if (Z_ISUNDEF(obj_data)) {
 					/* zend_call_method() failed */
-					break;
+					return;
```

Comment:
With the original `break`, this would have returned in the block I removed below since `ht_data` is not initialized at this point. If `zend_call_method()` fails, we either had an exception from `bsonSerialize()` or a PHP error from `zend_call_method()` itself. In any event, the behavior is the same in that we're not throwing any additional here.